contracts/signatures/AssetNonceVerifiable.sol

```diff
@@ -4,54 +4,69 @@ pragma solidity ^0.8.7;
-    /// @notice Current nonce per asset per signer.
+abstract contract AssetVerificationIndex is ContextUpgradeable {
+    /// @notice Current verification index per asset per signer.
     /// Updating it will invalidate all signatures created with the previous value on an asset level.
-    /// @custom:schema (contract address -> token id -> signer address -> nonce)
-    mapping(address => mapping(uint256 => mapping(address => uint256))) private assetNonce;
+    /// @custom:schema (contract address -> token id -> signer address -> verification index)
+    mapping(address => mapping(uint256 => mapping(address => uint256))) private assetVerificationIndex;
 
-    event AssetNonceUpdated(address indexed _signer, address indexed _contractAddress, uint256 indexed _tokenId, uint256 _newNonce, address _sender);
+    event AssetVerificationIndexUpdated(
+        address indexed _signer,
+        address indexed _contractAddress,
+        uint256 indexed _tokenId,
+        uint256 _newIndex,
+        address _sender
+    );
 
-    function __AssetNonceVerifiable_init() internal onlyInitializing {}
+    function __AssetVerificationIndex_init() internal onlyInitializing {}
 
-    function __AssetNonceVerifiable_init_unchained() internal onlyInitializing {}
+    function __AssetVerificationIndex_init_unchained() internal onlyInitializing {}
 
-    /// @notice Get the signer nonce for a given ERC721 token.
+    /// @notice Get the signer verification index for a given ERC721 token.
     /// @param _contractAddress The address of the ERC721 contract.
     /// @param _tokenId The id of the ERC721 token.
     /// @param _signer The address of the signer.
-    /// @return The nonce of the given signer for the provided asset.
-    function getAssetNonce(
+    /// @return The verification index of the given signer for the provided asset.
+    function getAssetVerificationIndex(
         address _contractAddress,
         uint256 _tokenId,
         address _signer
     ) external view returns (uint256) {
-        return assetNonce[_contractAddress][_tokenId][_signer];
+        return assetVerificationIndex[_contractAddress][_tokenId][_signer];
     }
 
-    /// @notice Increase the asset nonce of the sender by 1.
+    /// @notice Increase the asset verification index of the sender by 1.
     /// @param _contractAddress The contract address of the asset.
     /// @param _tokenId The token id of the asset.
-    function bumpAssetNonce(address _contractAddress, uint256 _tokenId) external {
-        _bumpAssetNonce(_contractAddress, _tokenId, _msgSender());
+    function bumpAssetVerificationIndex(address _contractAddress, uint256 _tokenId) external {
+        _bumpAssetVerificationIndex(_contractAddress, _tokenId, _msgSender());
     }
 
-    /// @dev Increase the asset nonce by 1
-    function _bumpAssetNonce(
+    /// @dev Increase the asset verification index by 1
+    function _bumpAssetVerificationIndex(
```

Comment:
Is there any reason why you have named it that way instead of `_bumpAssetIndex`?